app/models/user.rb

```diff
@@ -1,6 +1,41 @@
 class User < ActiveRecord::Base
+  before_create :grow_beard
+  before_save :hash_password
 
   def fullname
     first_name + " " + last_name
   end
+
+  def trim
+    self.status = "Trimmed"
+    self.save
+  end
+
+  def massage_and_trim
+    self.status = "Massaged and Trimmed"
+    self.save
+  end
+
+  def shave
+    self.status = "Shaved"
+    self.save
+  end
+
+  def grow_beard
+    self.status = "Unshaved beard"
+  end
+
+  def has_beard?
+    if self.status == "Shaved"
+      if((Time.now - self.updated_at) / 60 / 60) < 24
+        return false
+      end
+    end
+    return true
+  end
+
+  def hash_password
+    self.password = Digest::MD5.hexdigest(self.password)
```

Comment:
That was a nice article. Also, the password should be hashed in the client level itself and can be salted before storing, but since there is no authentication involved I didn't implemented that, but just a callback.